download-basket/src/main/java/downloadbasket/helpers/OGCServices.java

```diff
@@ -0,0 +1,208 @@
+package downloadbasket.helpers;
+
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.util.PropertyUtil;
+import downloadbasket.data.NormalWayDownloads;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.StringWriter;
+import java.net.URLEncoder;
+
+
+/**
+ * Created by markokuo on 6.10.2015.
+ */
+public class OGCServices {
+    private static final fi.nls.oskari.log.Logger LOGGER = LogFactory.getLogger(OGCServices.class);
+
+    private static final String PARAM_CROPPING_MODE = "croppingMode";
+    private static final String PARAM_CROPPING_LAYER = "croppingLayer";
+    private static final String PARAM_BBOX = "bbox";
+    private static final String PARAM_BBOX_LEFT = "left";
+    private static final String PARAM_BBOX_BOTTOM = "bottom";
+    private static final String PARAM_BBOX_RIGHT = "right";
+    private static final String PARAM_BBOX_TOP = "top";
+    private static final String PARAM_IDENTIFIERS = "identifiers";
+    private static final String PARAM_LAYER = "layer";
+
+    /**
+     * Get filter
+     * @param download download details
+     * @param writeParam write param
+     *
+     * @return filter url param and value
+     */
+    public static String getFilter(JSONObject download, Boolean writeParam) throws JSONException{
+        StringWriter s = new StringWriter();
+
+        String normalWayDownload = PropertyUtil.get("hsy.wfs.download.normal.way.downloads");
+        String[] temp = normalWayDownload.split(",");
+        NormalWayDownloads normalDownloads = new NormalWayDownloads();
+        for (int i = 0; i < temp.length; i++) {
```

Comment:
Use the enhanced format  for looping instead:

	for(String download: normalWayDownload.split(",")) {
		//  ...
	}

Reply:
fixed